src/autoload/Pyegi.lua

```diff
@@ -269,24 +384,51 @@ function Main(sub, sel)
 				l2.margin_t = tonumber(all_lines[line_params_number*(counter1-1) + 9])
 				l2.effect = all_lines[line_params_number*(counter1-1) + 10]
 				l2.text = all_lines[line_params_number*(counter1-1) + 11]
-				sub.insert(sel[line_number]+1, l2)
+				sub.insert(desired_lines_modified[line_number]+1, l2)
 			end
+			APT("")
 		end
-	elseif btn == "Settings" then
-		open_settings()
+	elseif btn == btns[2] then -- "Settings"
```

Comment:
I think "Settings" is better than btns[2]. because it avoids further difficulties in changing the order of the list. Think about a situation that we want to have another button before "Settings". We must change all btns[2] references to btns[3]. (خسته شدم انقدر انگلیسی نوشتم :)) این تیکه رو فارسی می‌گم. هرچند خود استرینگ هاردکد کردن هم بدی‌های خودش رو داره و بهترین حالت داشتن یه تیبل مشابه Enum توی زبانای دیگه‌ست. ولی خب اون رو ولش لازم نیست به نظرم فعلا تا اون حد آدم پیش بره و اصلا کد رو یکم کثیف می‌کنه احتمالا اون حالت. البته نمی‌دونم اصلا چقدر خوب بشه انجامش داد به هرحال... هدف اینه که آدم کمترین میزان کد رو تغییر بده و بتونه از خطا در رانتایم جلوگیری کنه. در حقیقت خطایی اگه لازم باشه، توی کامپایل‌تایم به آدم داده بشه)
the reason that we used btns[1] for the "Apply" was that we had an agreement between ourselves that the first and the last (btns[-1]) always will be "OK, Apply, etc" and "Cancel, Exit, etc". Or I deducted this from your code at first 😂.